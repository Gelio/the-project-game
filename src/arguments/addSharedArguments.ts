import { ArgumentParser } from 'argparse';

export function addSharedArguments(parser: ArgumentParser) {
  parser.addArgument(['-v', '--verbose'], { nargs: 0 });
  parser.addArgument(['--log-level'], {
    choices: ['error', 'warn', 'info', 'verbose', 'debug', 'silly']
  });
}
