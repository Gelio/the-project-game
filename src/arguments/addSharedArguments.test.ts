import { ArgumentParser } from 'argparse';
import { addSharedArguments } from './addSharedArguments';

describe('addSharedArguments', () => {
  let parser: ArgumentParser;

  beforeEach(() => {
    parser = <any>{
      addArgument: jest.fn()
    };
  });

  it('should match snapshot', () => {
    addSharedArguments(parser);

    expect((<jest.Mock>parser.addArgument).mock.calls).toMatchSnapshot();
  });
});
