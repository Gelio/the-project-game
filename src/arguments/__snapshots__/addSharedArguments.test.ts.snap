// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addSharedArguments should match snapshot 1`] = `
Array [
  Array [
    Array [
      "-v",
      "--verbose",
    ],
    Object {
      "nargs": 0,
    },
  ],
  Array [
    Array [
      "--log-level",
    ],
    Object {
      "choices": Array [
        "error",
        "warn",
        "info",
        "verbose",
        "debug",
        "silly",
      ],
    },
  ],
]
`;
